### PR TITLE
Countries should appear as dropdowns in volunteer search and sign up forms #454

### PR DESCRIPTION
**Pull request by Godsownchild**
issue solved #454 
Upon Volunteer sign up volunteers can now select countries and cities. But django superuser has to add countries and cities.
Question: Can i do it in such a way that administrators instead of django superuser can create countries and upon selection of country by volunteer the city list should be only cities in that country?

### CODE REVIEW

File: vms/registration/templates/registration/signup_volunteer.html [hunk 1]
```diff
@@ -214,7 +214,18 @@
                     {% trans "City" %}<span class="asteriskField">*</span>
                 </label>
                 <div class="controls">
+<!--
```
Comment (tapasweni-pathak) [inline vms/registration/templates/registration/signup_volunteer.html]: You can simply remove this. Git is there, every line we can figure out later if we need. :)

File: vms/registration/templates/registration/signup_volunteer.html [hunk 2]
```diff
@@ -214,7 +214,18 @@
...
                     <input id="id_city" class="textinput textInput form-control" type="text" placeholder="{% blocktrans %}City{% endblocktrans %}" name="vol-city" value="{% if volunteer_form.city.value %}{{ volunteer_form.city.value }}{% endif %}">
+-->
```
Comment (tapasweni-pathak) [inline vms/registration/templates/registration/signup_volunteer.html]: delete.

File: vms/registration/templates/registration/signup_volunteer.html [hunk 3]
```diff
@@ -266,10 +277,22 @@
                     {% trans "Country" %}<span class="asteriskField">*</span>
                 </label>
                 <div class="controls">
-                    <input id="id_country" class="textinput textInput form-control" type="text" placeholder="{% blocktrans %}Country{% endblocktrans %}" name="vol-country" value="{% if volunteer_form.country.value %}{{ volunteer_form.country.value }}{% endif %}">
+		<!--
+                    <input id="id_country" class="textinput textInput form-control" type="text" placeholder="{% blocktrans %}Country{% endblocktrans %}"  value="{% if volunteer_form.country.value %}{{ volunteer_form.country.value }}{% endif %}">-->
```
Comment (tapasweni-pathak) [inline vms/registration/templates/registration/signup_volunteer.html]: delete.

File: vms/registration/views.py [hunk 4]
```diff
@@ -8,6 +8,8 @@
 from django.core.urlresolvers import reverse_lazy
 from django.utils.decorators import method_decorator
 from administrator.forms import AdministratorForm
+#django-cities-light
```
Comment (tapasweni-pathak) [inline vms/registration/views.py]: delete.

File: vms/registration/views.py [hunk 5]
```diff
@@ -8,6 +8,8 @@
...
+from cities_light.models import City,Country
```
Comment (tapasweni-pathak) [inline vms/registration/views.py]: flake8.

File: vms/registration/views.py [hunk 6]
```diff
@@ -113,13 +115,19 @@ class VolunteerSignupView(TemplateView):
     def get(self, request):
         user_form = UserForm(prefix="usr")
         volunteer_form = VolunteerForm(prefix="vol")
+        #added django-cities-light
+        cities=City.objects.all()
+        countries=Country.objects.all()
```
Comment (tapasweni-pathak) [inline vms/registration/views.py]: flake8.

File: vms/volunteer/models.py [hunk 7]
```diff
@@ -35,14 +35,16 @@ class Volunteer(models.Model):
...
-            RegexValidator(
-                r'^[(A-Z)|(a-z)|(\s)|(\-)]+$',
-            ),
-        ],
-    )
+    #city = models.CharField(
+        #max_length=75,
+        #validators=[
+            #RegexValidator(
+                #r'^[(A-Z)|(a-z)|(\s)|(\-)]+$',
```
Comment (tapasweni-pathak) [inline vms/volunteer/models.py]: after removing the validator, we won't be allowing user to add a new country, is it?

File: vms/registration/views.py [hunk 8]
```diff
@@ -113,13 +115,19 @@ class VolunteerSignupView(TemplateView):
...
         return render(request,
                       'registration/signup_volunteer.html',
                       {'user_form': user_form,
                        'volunteer_form': volunteer_form,
                        'registered': self.registered,
                        'phone_error': self.phone_error,
                        'organization_list': self.organization_list,
+                       #added django-cities-light
```
Comment (tapasweni-pathak) [inline vms/registration/views.py]: same as above. :)

**Comment by Godsownchild** [comment]
Okay No problem

**Comment by tapasweni-pathak** [comment]
@Godsownchild were you able to fix this? Please let me know if you are stuck for long on something, we can see it together.

**Pull request closed without merging**